src/Native/Unix/System.Native/pal_io.c

```diff
@@ -1186,37 +1205,67 @@ static int32_t CopyFile_ReadWrite(int inFd, int outFd)
     free(buffer);
     return 0;
 }
-#endif // !HAVE_FCOPYFILE
 
-int32_t SystemNative_CopyFile(intptr_t sourceFd, intptr_t destinationFd)
+static int32_t CopyFileMetadata(int inFd, int outFd, struct stat_ *sourceStat)
```

Comment:
Why does CopyFileMetadata need to take a pointer to a `struct stat_` rather than just declaring one itself?  None of the call sites use the passed in pointer after the call, and I don't think the benefit of avoiding the extra local outweighs the cognitive cost of passing around pointers unnecessarily.